best-practices/client-side/framework/react/components/media-cropper/src/components/media-cropper/index.js

```diff
@@ -0,0 +1,204 @@
+            },
+            moved: true
+        });
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    mouseUp(e) {
+        this.setState({
+            ...this.state,
+            dragging: false
+        });
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    mouseWheel(e) {
+        const scale = Math.pow(1.0015, -e.deltaY);
+
+        const mX = e.pageX - e.target.offsetLeft;
+        const mY = e.pageY - e.target.offsetTop;
+
+        let transX = (this.state.transX - mX) * scale + mX;
+        let transY = (this.state.transY - mY) * scale + mY;
+
+        const zoom = this.state.zoom * scale;
+        this.setState({
+            ...this.state,
+            zoom,
+            transX,
+            transY
+        });
+    }
+
+    loadFile(e) {
+        e.preventDefault();
+        e.stopPropagation();
+        const file = e.target.files[0];
+        if (!file) return;
+        if (file.type === 'image/gif') {
+            // TODO: Handle gif
+        } else if (file.type.startsWith('image/')) {
+            const reader = new FileReader();
+
+            reader.onload = () => {
+                const media = new Image();
+                media.src = reader.result;
+                media.setAttribute('crossorigin', 'anonymous');
+
+                media.onload = () => {
+                    const zoom = Math.max(this.props.width / media.width, this.props.height / media.height);
+                    this.setState({
+                        ...this.state,
+                        isVideo: false,
+                        media,
+                        transX: 0,
+                        transY: 0,
+                        zoom,
+                        renderer: ImageRenderer
+                    });
+                    this.forceUpdate();
+                };
+            };
+
+            reader.readAsDataURL(file);
+        } else if (file.type.startsWith('video/')) {
+            const media = document.createElement('video');
+            media.muted = true;
+            media.src = URL.createObjectURL(file);
+            media.loop = true;
+            media.setAttribute('crossorigin', 'anonymous');
+            media.play();
+
+            media.onloadeddata = () => {
+                if (media.readyState >= 2) {
+                    media.width = media.videoWidth;
+                    media.height = media.videoHeight;
+                    const zoom = Math.max(this.props.width / media.width, this.props.height / media.height);
+                    media.onloadeddata = null;
+                    this.setState({
+                        ...this.state,
+                        isVideo: true,
+                        renderer: VideoRenderer,
+                        media,
+                        transX: 0,
+                        transY: 0,
+                        zoom
+                    });
+                }
+            }
+        } else {
+            console.log(file.type);
```

Comment:
I would remove all logs before this gets merged, or add `TODO`s to remove them in production.